ratelimit/redis.go

```diff
@@ -243,21 +245,26 @@ func (c *clusterLimitRedis) AllowContext(ctx context.Context, clearText string)
 		return false
 	}
 
-	pipe := c.ring.TxPipeline()
```

Comment:
What is the point of dropping pipeline? Would not it double the number of roundtrips?

Reply:
Great question!
The problem was, that this created socket writes from client side to a socket which was already half closed, so the Linux kernel which hosted redis just sent back RSTs.
So TxPipeline is network unsafe.

Reply:
As a note, the ring implementation [has changed quite a lot](https://github.com/go-redis/redis/compare/v7.0.0-beta.4...v8.3.3#diff-6b4fe479e7ce902cf98e5e4657ad83cba6afca1cd1703d99fd0f446dd53081a0)